### PR TITLE
Fix devicetree warnings

### LINKED ISSUE #29713: nRF5340 - duplicate unit-address
**Describe the bug**
During building, the following warnings are printed:

```
nrf5340dk_nrf5340_cpuapp.dts.pre.tmp:54.42-66.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@50000000/flash-controller@39000: duplicate unit-address (also used in node /soc/peripheral@50000000/kmu@39000)

nrf5340dk_nrf5340_cpuapp.dts.pre.tmp:328.19-334.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@50000000/clock@5000: duplicate unit-address (also used in node /soc/peripheral@50000000/power@5000)

nrf5340dk_nrf5340_cpuapp.dts.pre.tmp:584.31-585.5: Warning (unique_unit_address_if_enabled): /reserved-memory/image@20000000: duplicate unit-address (also used in node /reserved-memory/image_s@20000000)
```

**To Reproduce**
Compile any sample such as `samples/basic/button` using `west build -b nrf5340dk_nrf5340_cpuapp`

**Environment (please complete the following information):**
 - Linux
 - gcc-arm-none-eabi-9-2019-q4-major
 - b3ff89bd511afef484c193a428633f6a1c866944

### LINKED ISSUE #31947: Cleanup devicetree warnings generated by dtc
Cleanup the following warnings:

```
twister-out/96b_stm32_sensor_mez/samples/hello_world/sample.basic.helloworld/build.log:96b_stm32_sensor_mez.dts.pre.tmp:404.22-413.5: Warning (unique_unit_address_if_enabled): /soc/spi@40003800: duplicate unit-address (also used in node /soc/i2s@40003800)
twister-out/actinius_icarus_ns/samples/hello_world/sample.basic.helloworld/build.log:actinius_icarus_ns.dts.pre.tmp:58.42-70.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@40000000/flash-controller@39000: duplicate unit-address (also used in node /soc/peripheral@40000000/kmu@39000)
twister-out/actinius_icarus_ns/samples/hello_world/sample.basic.helloworld/build.log:actinius_icarus_ns.dts.pre.tmp:310.19-316.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@40000000/clock@5000: duplicate unit-address (also used in node /soc/peripheral@40000000/power@5000)
twister-out/actinius_icarus/samples/hello_world/sample.basic.helloworld/build.log:actinius_icarus.dts.pre.tmp:58.42-70.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@50000000/flash-controller@39000: duplicate unit-address (also used in node /soc/peripheral@50000000/kmu@39000)
twister-out/actinius_icarus/samples/hello_world/sample.basic.helloworld/build.log:actinius_icarus.dts.pre.tmp:310.19-316.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@50000000/clock@5000: duplicate unit-address (also used in node /soc/peripheral@50000000/power@5000)
twister-out/adafruit_feather_m0_basic_proto/samples/hello_world/sample.basic.helloworld/build.log:adafruit_feather_m0_basic_proto.dts.pre.tmp:101.29-105.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004400: duplicate unit-address (also used in node /soc/gpio@41004400)
twister-out/adafruit_feather_m0_basic_proto/samples/hello_world/sample.basic.helloworld/build.log:adafruit_feather_m0_basic_proto.dts.pre.tmp:106.29-110.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004480: duplicate unit-address (also used in node /soc/gpio@41004480)
twister-out/adafruit_itsybitsy_m4_express/samples/hello_world/sample.basic.helloworld/build.log:adafruit_itsybitsy_m4_express.dt.pre.tmp:136.29-140.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41008000: duplicate unit-address (also used in node /soc/gpio@41008000)
twister-out/adafruit_itsybitsy_m4_express/samples/hello_world/sample.basic.helloworld/build.log:adafruit_itsybitsy_m4_express.dt.pre.tmp:141.29-145.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41008080: duplicate unit-address (also used in node /soc/gpio@41008080)
twister-out/adafruit_itsybitsy_m4_express/samples/hello_world/sample.basic.helloworld/build.log:adafruit_itsybitsy_m4_express.dt.pre.tmp:146.29-150.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41008100: duplicate unit-address (also used in node /soc/gpio@41008100)
twister-out/adafruit_itsybitsy_m4_express/samples/hello_world/sample.basic.helloworld/build.log:adafruit_itsybitsy_m4_express.dt.pre.tmp:151.29-155.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41008180: duplicate unit-address (also used in node /soc/gpio@41008180)
twister-out/adafruit_trinket_m0/samples/hello_world/sample.basic.helloworld/build.log:adafruit_trinket_m0.dts.pre.tmp:123.28-128.5: Warning (spi_bus_bridge): /soc/sercom@42000c00: node name for SPI buses should be 'spi'
twister-out/adafruit_trinket_m0/samples/hello_world/sample.basic.helloworld/build.log:adafruit_trinket_m0.dts.pre.tmp:101.29-105.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004400: duplicate unit-address (also used in node /soc/gpio@41004400)
twister-out/adafruit_trinket_m0/samples/hello_world/sample.basic.helloworld/build.log:adafruit_trinket_m0.dts.pre.tmp:106.29-110.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004480: duplicate unit-address (also used in node /soc/gpio@41004480)
twister-out/arduino_nano_33_iot/samples/hello_world/sample.basic.helloworld/build.log:arduino_nano_33_iot.dts.pre.tmp:129.28-134.5: Warning (spi_bus_bridge): /soc/sercom@42001000: node name for SPI buses should be 'spi'
twister-out/arduino_nano_33_iot/samples/hello_world/sample.basic.helloworld/build.log:arduino_nano_33_iot.dts.pre.tmp:101.29-105.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004400: duplicate unit-address (also used in node /soc/gpio@41004400)
twister-out/arduino_nano_33_iot/samples/hello_world/sample.basic.helloworld/build.log:arduino_nano_33_iot.dts.pre.tmp:106.29-110.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004480: duplicate unit-address (also used in node /soc/gpio@41004480)
twister-out/arduino_zero/samples/hello_world/sample.basic.helloworld/build.log:arduino_zero.dts.pre.tmp:101.29-105.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004400: duplicate unit-address (also used in node /soc/gpio@41004400)
twister-out/arduino_zero/samples/hello_world/sample.basic.helloworld/build.log:arduino_zero.dts.pre.tmp:106.29-110.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004480: duplicate unit-address (also used in node /soc/gpio@41004480)
twister-out/atsamd20_xpro/samples/hello_world/sample.basic.helloworld/build.log:atsamd20_xpro.dts.pre.tmp:101.29-105.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004400: duplicate unit-address (also used in node /soc/gpio@41004400)
twister-out/atsamd20_xpro/samples/hello_world/sample.basic.helloworld/build.log:atsamd20_xpro.dts.pre.tmp:106.29-110.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004480: duplicate unit-address (also used in node /soc/gpio@41004480)
twister-out/atsamd21_xpro/samples/hello_world/sample.basic.helloworld/build.log:atsamd21_xpro.dts.pre.tmp:101.29-105.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004400: duplicate unit-address (also used in node /soc/gpio@41004400)
twister-out/atsamd21_xpro/samples/hello_world/sample.basic.helloworld/build.log:atsamd21_xpro.dts.pre.tmp:106.29-110.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004480: duplicate unit-address (also used in node /soc/gpio@41004480)
twister-out/atsame54_xpro/samples/hello_world/sample.basic.helloworld/build.log:atsame54_xpro.dts.pre.tmp:136.29-140.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41008000: duplicate unit-address (also used in node /soc/gpio@41008000)
twister-out/atsame54_xpro/samples/hello_world/sample.basic.helloworld/build.log:atsame54_xpro.dts.pre.tmp:141.29-145.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41008080: duplicate unit-address (also used in node /soc/gpio@41008080)
twister-out/atsame54_xpro/samples/hello_world/sample.basic.helloworld/build.log:atsame54_xpro.dts.pre.tmp:146.29-150.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41008100: duplicate unit-address (also used in node /soc/gpio@41008100)
twister-out/atsame54_xpro/samples/hello_world/sample.basic.helloworld/build.log:atsame54_xpro.dts.pre.tmp:151.29-155.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41008180: duplicate unit-address (also used in node /soc/gpio@41008180)
twister-out/atsamr21_xpro/samples/hello_world/sample.basic.helloworld/build.log:atsamr21_xpro.dts.pre.tmp:141.28-146.5: Warning (spi_bus_bridge): /soc/sercom@42001800: node name for SPI buses should be 'spi'
twister-out/atsamr21_xpro/samples/hello_world/sample.basic.helloworld/build.log:atsamr21_xpro.dts.pre.tmp:101.29-105.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004400: duplicate unit-address (also used in node /soc/gpio@41004400)
twister-out/atsamr21_xpro/samples/hello_world/sample.basic.helloworld/build.log:atsamr21_xpro.dts.pre.tmp:106.29-110.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004480: duplicate unit-address (also used in node /soc/gpio@41004480)
twister-out/atsamr21_xpro/samples/hello_world/sample.basic.helloworld/build.log:atsamr21_xpro.dts.pre.tmp:221.24-227.5: Warning (unique_unit_address_if_enabled): /soc/gpio@41004500: duplicate unit-address (also used in node /soc/pinmux@41004500)
twister-out/bcm958402m2_a72/samples/hello_world/sample.basic.helloworld/build.log:bcm958402m2_a72.dts.pre.tmp:99.21-111.5: Warning (simple_bus_reg): /soc/timer: missing or empty reg/ranges property
twister-out/bl652_dvk/samples/hello_world/sample.basic.helloworld/build.log:bl652_dvk.dts.pre.tmp:105.22-113.5: Warning (unique_unit_address_if_enabled): /soc/i2c@40003000: duplicate unit-address (also used in node /soc/spi@40003000)
twister-out/bl654_dvk/samples/hello_world/sample.basic.helloworld/build.log:bl654_dvk.dts.pre.tmp:124.22-132.5: Warning (unique_unit_address_if_enabled): /soc/i2c@40003000: duplicate unit-address (also used in node /soc/spi@40003000)
twister-out/circuitdojo_feather_nrf9160ns/samples/hello_world/sample.basic.helloworld/build.log:circuitdojo_feather_nrf9160ns.dt.pre.tmp:58.42-70.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@40000000/flash-controller@39000: duplicate unit-address (also used in node /soc/peripheral@40000000/kmu@39000)
twister-out/circuitdojo_feather_nrf9160ns/samples/hello_world/sample.basic.helloworld/build.log:circuitdojo_feather_nrf9160ns.dt.pre.tmp:310.19-316.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@40000000/clock@5000: duplicate unit-address (also used in node /soc/peripheral@40000000/power@5000)
twister-out/circuitdojo_feather_nrf9160/samples/hello_world/sample.basic.helloworld/build.log:circuitdojo_feather_nrf9160.dts.pre.tmp:58.42-70.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@50000000/flash-controller@39000: duplicate unit-address (also used in node /soc/peripheral@50000000/kmu@39000)
twister-out/circuitdojo_feather_nrf9160/samples/hello_world/sample.basic.helloworld/build.log:circuitdojo_feather_nrf9160.dts.pre.tmp:310.19-316.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@50000000/clock@5000: duplicate unit-address (also used in node /soc/peripheral@50000000/power@5000)
twister-out/efr32mg_sltb004a/samples/hello_world/sample.basic.helloworld/build.log:efr32mg_sltb004a.dts.pre.tmp:76.26-84.5: Warning (spi_bus_bridge): /soc/usart@40010800: node name for SPI buses should be 'spi'
twister-out/efr32_radio_brd4104a/samples/hello_world/sample.basic.helloworld/build.log:efr32_radio_brd4104a.dts.pre.tmp:66.26-74.5: Warning (spi_bus_bridge): /soc/usart@40010400: node name for SPI buses should be 'spi'
twister-out/efr32_radio_brd4250b/samples/hello_world/sample.basic.helloworld/build.log:efr32_radio_brd4250b.dts.pre.tmp:66.26-74.5: Warning (spi_bus_bridge): /soc/usart@40010400: node name for SPI buses should be 'spi'
twister-out/faze/samples/hello_world/sample.basic.helloworld/build.log:faze.dts.pre.tmp:48.18-50.5: Warning (unique_unit_address_if_enabled): /soc/flash@0: duplicate unit-address (also used in node /soc/gpio@0)
twister-out/lpcxpresso11u68/samples/hello_world/sample.basic.helloworld/build.log:lpcxpresso11u68.dts.pre.tmp:48.18-50.5: Warning (unique_unit_address_if_enabled): /soc/flash@0: duplicate unit-address (also used in node /soc/gpio@0)
twister-out/lpcxpresso55s16_ns/samples/hello_world/sample.basic.helloworld/build.log:lpcxpresso55s16_ns.dts.pre.tmp:81.19-86.4: Warning (unique_unit_address_if_enabled): /soc/peripheral@40000000/syscon@0: duplicate unit-address (also used in node /soc/peripheral@40000000/gpio@0)
twister-out/lpcxpresso55s28_ns/samples/hello_world/sample.basic.helloworld/build.log:lpcxpresso55s28_ns.dts.pre.tmp:87.19-92.4: Warning (unique_unit_address_if_enabled): /soc/peripheral@40000000/syscon@0: duplicate unit-address (also used in node /soc/peripheral@40000000/gpio@0)
twister-out/lpcxpresso55s28/samples/hello_world/sample.basic.helloworld/build.log:lpcxpresso55s28.dts.pre.tmp:115.19-120.4: Warning (unique_unit_address_if_enabled): /soc/peripheral@50000000/syscon@0: duplicate unit-address (also used in node /soc/peripheral@50000000/gpio@0)
twister-out/lpcxpresso55s69_cpu0/samples/hello_world/sample.basic.helloworld/build.log:lpcxpresso55s69_cpu0.dts.pre.tmp:123.19-128.4: Warning (unique_unit_address_if_enabled): /soc/peripheral@50000000/syscon@0: duplicate unit-address (also used in node /soc/peripheral@50000000/gpio@0)
twister-out/lpcxpresso55s69_cpu1/samples/hello_world/sample.basic.helloworld/build.log:lpcxpresso55s69_cpu1.dts.pre.tmp:95.19-100.4: Warning (unique_unit_address_if_enabled): /soc/peripheral@40000000/syscon@0: duplicate unit-address (also used in node /soc/peripheral@40000000/gpio@0)
twister-out/lpcxpresso55s69_ns/samples/hello_world/sample.basic.helloworld/build.log:lpcxpresso55s69_ns.dts.pre.tmp:95.19-100.4: Warning (unique_unit_address_if_enabled): /soc/peripheral@40000000/syscon@0: duplicate unit-address (also used in node /soc/peripheral@40000000/gpio@0)
twister-out/nrf9160dk_nrf9160ns/samples/hello_world/sample.basic.helloworld/build.log:nrf9160dk_nrf9160ns.dts.pre.tmp:58.42-70.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@40000000/flash-controller@39000: duplicate unit-address (also used in node /soc/peripheral@40000000/kmu@39000)
twister-out/nrf9160dk_nrf9160ns/samples/hello_world/sample.basic.helloworld/build.log:nrf9160dk_nrf9160ns.dts.pre.tmp:310.19-316.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@40000000/clock@5000: duplicate unit-address (also used in node /soc/peripheral@40000000/power@5000)
twister-out/nrf9160dk_nrf9160/samples/hello_world/sample.basic.helloworld/build.log:nrf9160dk_nrf9160.dts.pre.tmp:58.42-70.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@50000000/flash-controller@39000: duplicate unit-address (also used in node /soc/peripheral@50000000/kmu@39000)
twister-out/nrf9160dk_nrf9160/samples/hello_world/sample.basic.helloworld/build.log:nrf9160dk_nrf9160.dts.pre.tmp:310.19-316.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@50000000/clock@5000: duplicate unit-address (also used in node /soc/peripheral@50000000/power@5000)
twister-out/nrf9160_innblue21ns/samples/hello_world/sample.basic.helloworld/build.log:nrf9160_innblue21ns.dts.pre.tmp:58.42-70.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@40000000/flash-controller@39000: duplicate unit-address (also used in node /soc/peripheral@40000000/kmu@39000)
twister-out/nrf9160_innblue21ns/samples/hello_world/sample.basic.helloworld/build.log:nrf9160_innblue21ns.dts.pre.tmp:310.19-316.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@40000000/clock@5000: duplicate unit-address (also used in node /soc/peripheral@40000000/power@5000)
twister-out/nrf9160_innblue21/samples/hello_world/sample.basic.helloworld/build.log:nrf9160_innblue21.dts.pre.tmp:58.42-70.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@50000000/flash-controller@39000: duplicate unit-address (also used in node /soc/peripheral@50000000/kmu@39000)
twister-out/nrf9160_innblue21/samples/hello_world/sample.basic.helloworld/build.log:nrf9160_innblue21.dts.pre.tmp:310.19-316.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@50000000/clock@5000: duplicate unit-address (also used in node /soc/peripheral@50000000/power@5000)
twister-out/nrf9160_innblue22ns/samples/hello_world/sample.basic.helloworld/build.log:nrf9160_innblue22ns.dts.pre.tmp:58.42-70.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@40000000/flash-controller@39000: duplicate unit-address (also used in node /soc/peripheral@40000000/kmu@39000)
twister-out/nrf9160_innblue22ns/samples/hello_world/sample.basic.helloworld/build.log:nrf9160_innblue22ns.dts.pre.tmp:310.19-316.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@40000000/clock@5000: duplicate unit-address (also used in node /soc/peripheral@40000000/power@5000)
twister-out/nrf9160_innblue22/samples/hello_world/sample.basic.helloworld/build.log:nrf9160_innblue22.dts.pre.tmp:58.42-70.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@50000000/flash-controller@39000: duplicate unit-address (also used in node /soc/peripheral@50000000/kmu@39000)
twister-out/nrf9160_innblue22/samples/hello_world/sample.basic.helloworld/build.log:nrf9160_innblue22.dts.pre.tmp:310.19-316.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@50000000/clock@5000: duplicate unit-address (also used in node /soc/peripheral@50000000/power@5000)
twister-out/qemu_cortex_a53/samples/hello_world/sample.basic.helloworld/build.log:qemu_cortex_a53.dts.pre.tmp:60.21-71.5: Warning (simple_bus_reg): /soc/timer: missing or empty reg/ranges property
twister-out/sam4l_ek/samples/hello_world/sample.basic.helloworld/build.log:sam4l_ek.dts.pre.tmp:311.28-315.5: Warning (simple_bus_reg): /soc/device_uid@0080020c: simple-bus unit address format error, expected "80020c"
twister-out/seeeduino_xiao/samples/hello_world/sample.basic.helloworld/build.log:seeeduino_xiao.dts.pre.tmp:101.29-105.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004400: duplicate unit-address (also used in node /soc/gpio@41004400)
twister-out/seeeduino_xiao/samples/hello_world/sample.basic.helloworld/build.log:seeeduino_xiao.dts.pre.tmp:106.29-110.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004480: duplicate unit-address (also used in node /soc/gpio@41004480)
twister-out/serpente/samples/hello_world/sample.basic.helloworld/build.log:serpente.dts.pre.tmp:101.29-105.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004400: duplicate unit-address (also used in node /soc/gpio@41004400)
twister-out/serpente/samples/hello_world/sample.basic.helloworld/build.log:serpente.dts.pre.tmp:106.29-110.5: Warning (unique_unit_address_if_enabled): /soc/pinmux@41004480: duplicate unit-address (also used in node /soc/gpio@41004480)
twister-out/udoo_neo_full_m4/samples/hello_world/sample.basic.helloworld/build.log:udoo_neo_full_m4.dts.pre.tmp:71.26-74.4: Warning (unit_address_format): /memory@0x84000000: unit name should not have leading "0x"
twister-out/udoo_neo_full_m4/samples/hello_world/sample.basic.helloworld/build.log:udoo_neo_full_m4.dts.pre.tmp:75.25-77.4: Warning (unit_address_format): /memory@0x84080000: unit name should not have leading "0x"
twister-out/v2m_musca_b1_nonsecure/samples/hello_world/sample.basic.helloworld/build.log:v2m_musca_b1_nonsecure.dts.pre.tmp:55.25-57.4: Warning (unit_address_format): /flash@0a070000: unit name should not have leading 0s
```

### PR DESCRIPTION
**Pull request by galak**
Fixes #31947 and Fixes #29713

### CODE REVIEW

**Comment by galak** [comment]
This includes the fix in PR #31950 but accomplishes it for all boards and all warnings.

**Comment by galak** [comment]
Fixes #31947 and #29713

**Review by mbolivar-nordic — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Review by mbittan — APPROVED** [review]

**Review by simonguinot — APPROVED** [review]

File: boards/arm/adafruit_trinket_m0/pre_dt_board.cmake [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Copyright (c) 2021 Linaro Limited
+# SPDX-License-Identifier: Apache-2.0
+
+# SPI is implemented via sercom so node name isn't spi@...
+list(APPEND EXTRA_DTC_FLAGS "-Wno-spi_bus_bridge")
```
Comment (MaureenHelm) [inline boards/arm/adafruit_trinket_m0/pre_dt_board.cmake:5]: Do we need to do this for the flexcomm spi on lpcxpresso and mimxrt685 boards too?

Comment (galak) [reply boards/arm/adafruit_trinket_m0/pre_dt_board.cmake]: No, I'm guessing they covered by the fact we set "-Wno-unique_unit_address" globally in camek/dts.cmake covers them.

**Comment by galak** [comment]
Splitting this PR to hopefully get CI to pass.

**Pull request merged**